pkg/subctl/cmd/gather/summary.go

```diff
@@ -0,0 +1,224 @@
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// Embed the file content as string.
+//go:embed layout.gohtml
+var layout string
+
+type version struct {
+	Subctl    string
+	Subm      string
+	K8sServer string
+}
+
+type clusterConfig struct {
+	CNIPlugin        string
+	CloudProvider    string
+	TotalNode        int
+	GatewayNode      map[string]types.UID
+	GWNodeNumber     int
+	MasterNode       map[string]types.UID
+	MasterNodeNumber int
+}
+
+type nodeConfig struct {
+	Name string
+	Info v1.NodeSystemInfo
+}
+
+type logInfo struct {
+	PodName      string
+	Namespace    string
+	RestartCount int32
+	PodState     v1.PodPhase
+	LogFileName  []string
+}
+
+type resourceInfo struct {
+	Name      string
+	Namespace string
+	Type      string
+	FileName  string
+}
+
+type data struct {
+	ClusterName   string
+	Versions      version
+	ClusterConfig clusterConfig
+	NodeConfig    []nodeConfig
+	PodLogs       []logInfo
+	ResourceInfo  []resourceInfo
+}
+
+func ClusterInformation(info Info) {
+	dataGathered := getClusterInfo(info)
+	file := createFile(info.DirName)
+	WriteToHTML(file, dataGathered)
+	PodLogInfos = nil
+	Resources = nil
+}
+
+func getClusterInfo(info Info) data {
+	versions := getVersions(info)
+	config := getClusterConfig(info)
+	nConfig := getNodeConfig(info)
+	podLogs := populatePodLogInfo()
+	resourcesInfo := populateResourceInfo()
+	d := data{
+		ClusterName:   info.ClusterName,
+		Versions:      versions,
+		ClusterConfig: config,
+		NodeConfig:    nConfig,
+		PodLogs:       podLogs,
+		ResourceInfo:  resourcesInfo,
+	}
+	return d
+}
+
+func getClusterConfig(info Info) clusterConfig {
+	cniPlugin := "Not found"
+	if info.Submariner != nil {
+		cniPlugin = info.Submariner.Status.NetworkPlugin
+	}
+	gwNodes := getGWNodes(info)
+	mNodes := getMasterNodes(info)
+	allNodes, _ := listNodes(info, metav1.ListOptions{})
+	config := clusterConfig{
+		CNIPlugin:        cniPlugin,
+		CloudProvider:    "AWS", // TODO Get this from Submariner once https://github.com/submariner-io/submariner-operator/pull/1252 is merged
+		TotalNode:        len(allNodes.Items),
+		GatewayNode:      gwNodes,
+		GWNodeNumber:     len(gwNodes),
+		MasterNode:       mNodes,
+		MasterNodeNumber: len(mNodes),
+	}
+	return config
+}
+
+func getVersions(info Info) version {
+	k8sServerVersion, err := info.ClientSet.Discovery().ServerVersion()
+	serverVersion := k8sServerVersion.String()
+	if err != nil {
+		fmt.Println("error in getting k8s server version", err)
+		serverVersion = err.Error()
+	}
+
+	submVer := "Not installed"
+	if info.Submariner != nil {
+		submVer = info.Submariner.Spec.Version
+	}
+
+	Versions := version{
+		Subctl:    subctlversion.Version,
+		Subm:      submVer,
+		K8sServer: serverVersion,
+	}
+	return Versions
+}
+
+func getSpecificNode(info Info, selector string) map[string]types.UID {
+	nodes, err := listNodes(info, metav1.ListOptions{LabelSelector: selector})
+	if err != nil {
+		fmt.Println(err)
```

Comment:
Should return here.

Reply:
return `err`? this information is populated in html which currently does not handle errors. Errors just get printed to stdout

Reply:
I mean `return` so it doesn't possibly reference a nil `nodes` below.